modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaHttpServerCodegen.java

```diff
@@ -0,0 +1,486 @@
+    }
+
+
+
+    @Override
+    public Map<String, Object> postProcessOperationsWithModels(Map<String, Object> objs, List<Object> allModels) {
+        Map<String, Object> baseObjs = super.postProcessOperationsWithModels(objs, allModels);
+        pathMatcherPatternsPostProcessor(baseObjs);
+        marshallingPostProcessor(baseObjs);
+        return baseObjs;
+    }
+
+    private static Set<String> primitiveParamTypes = new HashSet<String>(){{
+        addAll(Arrays.asList(
+                "Int",
+                "Long",
+                "Float",
+                "Double",
+                "Boolean",
+                "String"
+        ));
+    }};
+
+    private static Map<String, String> pathTypeToMatcher = new HashMap<String, String>(){{
+        put("Int", "IntNumber");
+        put("Long", "LongNumber");
+        put("Float","FloatNumber");
+        put("Double","DoubleNumber");
+        put("Boolean","Boolean");
+        put("String", "Segment");
+    }};
+
+    protected static void addPathMatcher(CodegenOperation codegenOperation) {
+        LinkedList<String> allPaths = new LinkedList<>(Arrays.asList(codegenOperation.path.split("/")));
+        allPaths.removeIf(""::equals);
+
+        LinkedList<TextOrMatcher> pathMatchers = new LinkedList<>();
+        for(String path: allPaths){
+            TextOrMatcher textOrMatcher = new TextOrMatcher("", true, true);
+            if(path.startsWith("{") && path.endsWith("}")) {
+                String parameterName = path.substring(1, path.length()-1);
+                for(CodegenParameter pathParam: codegenOperation.pathParams){
+                    if(pathParam.baseName.equals(parameterName)) {
+                        String matcher = pathTypeToMatcher.get(pathParam.dataType);
+                        if(matcher == null) {
+                            LOGGER.warn("The path parameter " + pathParam.baseName +
+                                    " with the datatype " + pathParam.dataType +
+                                    " could not be translated to a corresponding path matcher of akka http" +
+                                    " and therefore has been translated to string.");
+                            matcher = pathTypeToMatcher.get("String");
+                        }
+                        if (pathParam.pattern != null && !pathParam.pattern.isEmpty()) {
+                            matcher = pathMatcherPatternName(pathParam);
+                        }
+                        textOrMatcher.value = matcher;
+                        textOrMatcher.isText = false;
+                        pathMatchers.add(textOrMatcher);
+                    }
+                }
+            } else {
+                textOrMatcher.value = path;
+                textOrMatcher.isText = true;
+                pathMatchers.add(textOrMatcher);
+            }
+        }
+        pathMatchers.getLast().hasMore = false;
+
+        codegenOperation.vendorExtensions.put("paths", pathMatchers);
```

Comment:
We've trying to normalize the naming of vendor extensions. I wonder if you can rename `paths` to `x-paths` instead.